Yank/yank.py

```diff
@@ -179,6 +183,66 @@ def ions_atoms(self):
                 if '-' in self._topology.atom(i).residue.name or
                 '+' in self._topology.atom(i).residue.name]
 
+    def add_region(self, region_name, region_selection):
+        """
+        Add a region to the Topography based on a selection string of atoms. The selection accepts multiple formats
+        such as a DSL string, a SMIRKS selection string, or hard coded atom indices. The selection string is converted
+        to a list of atom indices.
+
+        Parameters
+        ----------
+        region_name : str
+            Name of the region. This must be unique and also not the name of an existing method
+        region_selection : str or list of ints
+            Atom selection identifier, either a MDTraj DSL string, a SMIRKS string, or a hard-coded list of ints
+            The SMIRKS string requires the OpenEye OEChem library to correctly use
+            TODO: SMIRKS is not implemented in this build, but is planned
+
+        """
+        self._check_existing_regions(region_name)
+        atom_selection = self._resolve_atom_indices(region_selection)
+        self._regions[region_name] = atom_selection
+        pass
+
+    def remove_region(self, region_name):
+        """
+        Remove a previously added region from this Topography. This only affects regions added through the
+        :func:`add_region` function.
+
+        Does nothing if the region was not previously added
+
+        Parameters
+        ----------
+        region_name : str
+            Name of the region to remove
+
+        """
+        self._regions.pop(region_name, None)
+
+    def get_region(self, region_name):
+        """
+        Retrieve the atom indices of the given region. This function will also fetch the built-in regions
+
+        Parameters
+        ----------
+        region_name
+
+        Returns
+        -------
+
+        Raises
+        ------
+        KeyError
+            If region is not part of the Topography
+        """
+        if region_name not in self._combied_regions:
+            raise KeyError("Cannot find region \"{}\" in this Topography.".format(region_name))
```

Comment:
I think this `if...raise` should go after the check for built-in-regions below.

Reply:
Why? The `_combined_regions` property is the union of the `built-in` and the custom `regions`? I could make it it try to get the built in, then try to get the custom and raising an error if it can't find it.

Reply:
Oh, right! Sorry for the confusion.

Reply:
You could also use
```python
if region_name not in self:
```
since you implemented a `__contains__` function